contrib/cirrus/pr-should-include-tests

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+#
+# Intended for use in CI: check git commits, barf if no tests added.
+#
+
+# Docs-only changes are excused
+if [[ "${CIRRUS_CHANGE_TITLE}" =~ CI:DOCS ]]; then
+    exit 0
+fi
+
+# So are PRs where 'No test(s) needed' appears in the Github message
+if [[ "${CIRRUS_CHANGE_MESSAGE,,}" =~ no.tests?.needed ]]; then
+    exit 0
+fi
+
+# HEAD should be good enough, but the CIRRUS envariable allows us to test
+head=${CIRRUS_CHANGE_IN_REPO:-HEAD}
+# Base of this PR. Here we absolutely rely on cirrus.
+base=$(git merge-base ${DEST_BRANCH:-master} $head)
+
+# This gives us a list of files touched in all commits, e.g.
+#    A    foo.c
+#    M    bar.c
+# We look for Added or Modified (not Deleted!) files under 'test'.
+if git diff --name-status $base $head | egrep -q '^[AM]\s+(test/|.*_test\.go)'; then
+    exit 0
+fi
+
+# Nothing changed under test subdirectory.
+#
+# This is OK if the only files being touched are "safe" ones.
+filtered_changes=$(git diff --name-status $base $head |
+                       awk '{print $2}'               |
+                       fgrep -vx .cirrus.yml          |
+                       fgrep -vx go.mod               |
+                       fgrep -vx go.sum               |
+                       egrep -v  '^docs/'             |
+                       egrep -v  '^vendor/')
+if [[ -z "$filtered_changes" ]]; then
+    exit 0
+fi
+
+# One last chance: perhaps the developer included the magic 'no test needed'
+# string in an amended commit.
+if git log --format=%B ${base}..${head} | egrep -i 'no.tests?.needed'; then
+   exit 0
+fi
+
+cat <<EOF
+$(basename $0): PR does not include changes in the 'tests' directory
+
+Please write a regression test for what you're fixing. Even if it
+seems trivial or obvious, try to add a test that will prevent
+regressions.
+
+If your change is minor, feel free to piggyback on existing tests.
+Every second counts in CI.
+
+If your commit really, truly does not need tests, you can proceed
+by adding 'No test needed' or 'No tests needed' to the body of
```

Comment:
Soft suggestion.  I'd change the required string to "[NO TESTS NEEDED]".  Hopefully, it will make the submitter think a bit more, and it will more likely stand out in the description for reviewers.

Reply:
I like that. The brackets are going to be hard to check for cleanly, so I will look into it next week. Thank you!

Reply:
Done and tested.